pygeoapi/filters.py

```diff
@@ -0,0 +1,290 @@
+from operator import and_, or_, add, sub, mul, truediv
+# from datetime import datetime, timedelta
+# from functools import reduce
+
+
+def combine(sub_filters, combinator="AND"):
+    """ Combine filters using a logical combinator
+
+        :param sub_filters: the filters to combine
+        :param combinator: a string: "AND" / "OR"
+        :type sub_filters: list of filtered dict 
+        :return: the combined filter
+        :rtype: filtered dict
+    """
+    # check for sub_filters being the evaluation result of multiple sub_filters
+    assert isinstance(sub_filters, tuple)
+
+    # check for sub_filter being the list of filtered dataset
+    for sub_filter in sub_filters:
+        assert isinstance(sub_filter, list)
+
+    # check for appropriate type of combinator
+    assert combinator in ("AND", "OR")
+
+    mapping_list = []
+    intersection = []
+    union = []
+    
+    for row in sub_filters[0]:
+        if row in sub_filters[1]:
+            intersection.append(row)
+        else:
+            union.append(row)
+    
+    # perform combination operation
+    if combinator == "AND":
+        mapping_list=intersection
+    else:
+        mapping_list = union + sub_filters[1]
+        
+    return mapping_list
+
+
+def negate(sub_filter):  # TODO!!
+    """ Negate a filter, opposing its meaning.
+
+        :param sub_filter: the filter to negate
+        :type sub_filter: query
+        :return: the negated filter
+        :rtype: 
+    """
+
+# Comparison operators dictionary
+OP_TO_COMP = {
+    "<": "<",
+    "<=": "<=",
+    ">": ">",
+    ">=": ">=",
+    "<>": "!=",
+    "=": "=="
+}
+
+
+def compare(lhs, rhs, op, mapping_choices=None, field_mapping=None):
+    """ Compare a filter with an expression using a comparison operation
+
+        :param lhs: the field to compare
+        :type lhs: string
+        :param rhs: the filter expression
+        :type rhs: int/float
+        :param op: a string denoting the operation. one of ``"<"``, ``"<="``,
+                   ``">"``, ``">="``, ``"<>"``, ``"="``
+        :type op: str
+        :param mapping_choices: a list of dict to lookup potential choices for a certain
+                                field.
+        :type mapping_choices: list
+        :return: a comparison expression filter
+        :rtype: dict
+    """
+    # check for lhs to be a field name
+    assert lhs in field_mapping
+    # assert rhs  # TODO!!
+
+    # check for valid type of comparison operator
+    assert op in OP_TO_COMP
+    comp = OP_TO_COMP[op]
+
+    field_name = lhs
+
+    mapping_list=[]
+    if comp:
+        for row in mapping_choices:
+            # check for field to be present in dataset
+            assert lhs in row
+            
+            # perform comparison operation
+            if eval(row[lhs]+comp+str(rhs)):
+                mapping_list.append(row)
+
+    return mapping_list
+
+
+def between(lhs, low, high, not_=False): # TODO!!
```

Comment:
these functions below should have a `pass` at least or raise a `NotImplementedError`

Reply:
done